Import preliminary translations

vmarta opened the pull request:
# Description

Our translators have given us some preliminary translations. The final version will be coming later (after they test in context by visiting [this site](https://scaffold-pwa-translations.mobify-storefront.com/)). In the meantime, this PR would import those translations:
- There are 9 languages in total
- ~But on the website's locale selector dropdown, we're only showing a subset of them because some do not have the corresponding localized catalog yet in Business Manager~ Now we're showing all languages in the locale selector dropdown.

Ticket: https://gus.lightning.force.com/a07EE00000Tsky3YAB

# Types of Changes

<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->

- [ ] **Bug fix** (non-breaking change that fixes an issue)
- [x] **New feature** (non-breaking change that adds functionality)
- [ ] **Documentation update**
- [ ] **Breaking change** (could cause existing functionality to not work as expected)
- [ ] **Other changes** (non-breaking changes that does not fit any of the above)

> Breaking changes include:
>
> - Removing a public function or component or prop
> - Adding a required argument to a function
> - Changing the data type of a function parameter or return value
> - Adding a new peer dependency to `package.json`

# How to Test-Drive This PR

1. Visit either https://scaffold-pwa-translations.mobify-storefront.com/ or localhost
2. Go to the footer and select a different language/locale
3. Verify that most hardcoded text are no longer in English ← I write down "most" because there are some new hardcoded text on the homepage that the translators are not aware of

# Checklists

<!--- Enter an `x` in all the boxes that apply. -->
<!--- If you’re unsure about any of these, don’t hesitate to ask. We’re here to help! -->

## General

- [x] Changes are covered by test cases
- [x] CHANGELOG.md updated with a short description of changes (_not_ required for documentation updates)

## Accessibility Compliance

You must check off all items in **one** of the follow two lists:

- [ ] There are no changes to UI

_or..._

- [ ] Changes were tested with a Screen Reader (iOS VoiceOver or Android Talkback) and had no issues
- [ ] Changes comply with [WCAG 2.0 guidelines levels A and AA](https://www.wuhcag.com/wcag-checklist/)
- [ ] Changes to common UI patterns and interactions comply with [WAI-ARIA best practices](https://www.w3.org/TR/wai-aria-practices-1.1/)

## Localization

- [ ] Changes include a UI text update in the Retail React App (which requires translation)

vmarta committed:
Import preliminary translations

vmarta committed:
Add new locales

vmarta committed:
Revert "Add new locales"

This reverts commit 858636d9474e33ebc3b71c2e1ddb6f622c65b154.

vmarta committed:
No longer hardcoding 'Coming Soon'

vmarta committed:
Fix locale.js test

vmarta commented on packages/pwa/app/utils/locale.test.js:
```diff
@@ -22,7 +22,7 @@ const nonSupportedLocale = 'nl-NL'
 
 const testId1 = 'footer.link.privacy_policy'
-const testId2 = 'homepage.message.welcome'
+const testId2 = 'account.accordion.button.my_account'
```
`homepage.message.welcome` was just some sample id we used. Now that we have real translations, we can use real message ids too.

vmarta commented on packages/pwa/app/components/locale-text/index.jsx:
```diff
@@ -99,5 +99,5 @@ const LOCALE_MESSAGES = defineMessages({
     'zh-CN': {id: 'locale_text.message.zh-CN', defaultMessage: 'Chinese (China)'},
     'zh-HK': {id: 'locale_text.message.zh-HK', defaultMessage: 'Chinese (Hong Kong)'},
-    'zh-TW': {id: 'locale_text.message.zh-TW', defaultMessage: 'Chinese (Taiwan)'}
+    'zt-TW': {id: 'locale_text.message.zt-TW', defaultMessage: 'Chinese (Taiwan)'}
```
According to our translators, this is the correct locale id for Chinese Taiwan.

vmarta commented on packages/pwa/app/pages/product-detail/index.jsx:
```diff
@@ -220,7 +220,10 @@ const ProductDetail = ({category, product, isLoading}) => {
-                                Coming Soon
+                                {formatMessage({
+                                    defaultMessage: 'Coming Soon',
+                                    id: 'product_detail.accordion.message.coming_soon'
```
Oops, we forgot to format this hardcoded "Coming Soon".

vmarta committed:
Merge branch 'develop' into import-translations

vmarta committed:
Update CHANGELOG.md

vmarta committed:
Use zh-TW instead, as per BM config

vmarta committed:
Include locales we have translations for

And sort them by their ids.

vmarta committed:
Fix locale.js tests

vmarta committed:
Merge branch 'develop' into import-translations

vmarta commented on packages/pwa/package.json:
```diff
@@ -188,26 +188,16 @@
+            { "id": "it-IT", "preferredCurrency": "EUR" },
+            { "id": "ja-JP", "preferredCurrency": "JPY" },
+            { "id": "ko-KR", "preferredCurrency": "KRW" },
+            { "id": "pt-BR", "preferredCurrency": "BRL" },
```
There are new languages/locales that were not there before. To be able to show them properly on the website, I had to add new locales and currencies via Business Manager (BM). Thankfully, it was quite straightforward to do. I can show you if you want.

And yes, we still need to back up the site configs in BM. I'll make sure to back them up.

alexvuong replied:
Also, do you want to add these notes into the readme file?

vmarta replied:
@alexvuong Ok I've updated the translations readme:
- I saw that Vincent C had already written about something similar. So I just built on top of it.
- I took some time to clean up the readme too.

vmarta committed:
Merge branch 'develop' into import-translations

alexvuong commented on packages/pwa/package.json:
```diff
@@ -188,26 +188,16 @@
-                "id": "ja-JP",
-                "preferredCurrency": "JPY"
-            }
+            { "id": "de-DE", "preferredCurrency": "EUR" },
```
Looks like code format is different with the changes, it was supposed to create new line on the '{'. Maybe you want to check your prettier?

vmarta committed:
Format with prettier

vmarta committed:
Simplify the translations readme

- Make it clear that this is about translations and not the general term 'localization'
- Delete details that are no longer relevant (e.g. `defaultLocaleMessages`)
- Delete text with duplicate meanings (about the npm scripts)

vmarta committed:
Mention the locale's currency

vmarta committed:
More accurate title

vmarta committed:
Update what pseudo locale looks like

vmarta committed:
Merge branch 'develop' into import-translations

alexvuong commented on packages/pwa/app/translations/README.md:
```diff
@@ -43,84 +43,38 @@ Here’s the default locale configuration in `pwa/package.json`:
-
-We’ve provided the following npm scripts to help you:
-
--   `npm run build-translations` to extract and compile default messages into the default locale
```
Should we keep these texts to explain how the translation scripts are about?

alexvuong replied:
Oh I realized we mentioned it at the beginning of the file.

alexvuong reviewed (approved):
LGTM
